src/canonical.js

```diff
@@ -67,7 +67,9 @@ function toCanonical (form) {
         // 3.4.1.4. we return the output of applying the `consistency-check` algorithm to `items-type`
         return unionArray
       })
-      items.required = form.required
+      if (typeof form.required === 'boolean') {
+        items.required = form.required
```

Comment:
in what context would items have `required` set here? i.e. do we even need this line?

Reply:
There's a test failure if it is removed:

```
  1) canonicalForm() should generate canonical form of type Songs.Musician:

      AssertionError: expected { Object (type, additionalProperties, ...) } to deeply equal { Object (type, additionalProperties, ...) }
      + expected - actual

                     }
                   }
                   "type": "object"
                 }
      -          "required": [undefined]
      +          "required": true
                 "type": "array"
               }
               "name": {
                 "required": true
                     }
                   }
                   "type": "object"
                 }
      -          "required": [undefined]
      +          "required": true
                 "type": "array"
               }
               "name": {
                 "required": true
```

While it seems strange to set `required` on `items`, I think this must be used to propagate `required` when hoisting arrays of unions (to form a union of arrays).

Reply:
that makes sense. then let's keep it. will merge & release this today.